java/code/src/com/redhat/rhn/manager/channel/ChannelManager.java

```diff
@@ -188,10 +195,10 @@ public static void refreshWithNewestPackages(Channel channel, String label) {
      */
     public static void refreshWithNewestPackages(Long channelId, String label) {
         Channel chan = ChannelFactory.lookupById(channelId);
-         ChannelFactory.refreshNewestPackageCache(channelId, label);
-         if (chan != null) {
-             ChannelManager.queueChannelChange(chan.getLabel(), label, label);
-         }
+        ChannelFactory.refreshNewestPackageCache(channelId, label);
+        if (chan != null) {
```

Comment:
Just in case you want to have Kevin happier, replace the whole `if` statement with `Optional.ofNullable(chan).ifPresent(c -> ChannelManager.queueChannelChange(c.getLabel(), label, label));` :wink: